### PR TITLE
realitycapture nerfacto variable resolution

### PR DESCRIPTION
**Pull request by machenmusik**
As it turns out, #1650 fails because neither realitycapture processing nor nerfacto training handles variable resolution (i.e. images of different dimensions).

This PR fixes realitycapture processing for variable resolution, and changes nerfacto config to allow variable resolution.

I'm not really sure why nerfacto forces fixed resolution, as the change seems trivial; if there are performance optimizations to be done when all images are the same dimensions, IMO that should be handled elsewhere than hardcoding selection of datamanager for all uses of a config.

### CODE REVIEW

**Comment by tancik** [comment]
I think you raise a good point that we should default handle variable res data, however I don't think we should just default to the `VariableResDatamanger` for a few reasons.

+ It is likely slower and will take more memory
+ It has defaults that will likely degrade quality on existing models, specifically https://github.com/nerfstudio-project/nerfstudio/blob/2c79d9fe52ae2eef2887aa621ae701ef91da95b0/nerfstudio/data/datamanagers/variable_res_datamanager.py#L58-L65

What we should do is add this collate logic into the `VanillaDataManager` and use it when the data is variable res. We can then delete VariableResDataManager

**Comment by tancik** [comment]
I believe those default values were a holdover from when they used to be the default in the vanilla datamanager (pre nerfstudio release). They don't make sense to use unless you have a very large dataset that can't fit into memory.

**Comment by machenmusik** [comment]
I think this should be good now

File: nerfstudio/data/datamanagers/base_datamanager.py [hunk 1]
```diff
@@ -385,6 +410,16 @@ def __init__(
 
         self.train_dataset = self.create_train_dataset()
         self.eval_dataset = self.create_eval_dataset()
+
+        if self.train_dataparser_outputs is not None:
+            cameras = self.train_dataparser_outputs.cameras
+            if len(cameras) > 1:
+                for i in range(1, len(cameras)):
+                    if cameras[0].width != cameras[i].width or cameras[0].height != cameras[i].height:
+                        print("Variable resolution, using variable_res_collate")
```
Comment (tancik) [inline nerfstudio/data/datamanagers/base_datamanager.py:419]: Use `CONSOLE.print()`

**Review by tancik — COMMENTED** [review]
Thanks for making the changes!

File: nerfstudio/process_data/realitycapture_utils.py [hunk 2]
```diff
@@ -59,16 +59,7 @@ def realitycapture_to_json(
...
-        camera_label = name.rpartition(".")[0]
-        if camera_label in image_filename_map:
-            img = np.array(Image.open(output_dir / image_filename_map[camera_label]))
-            break
-
-    height, width, _ = img.shape
-
-    data["h"] = int(height)
-    data["w"] = int(width)
+    # Don't assume all images are the same dimensions
```
[the remark this answers is not in the document]
Comment (machenmusik) [reply nerfstudio/process_data/realitycapture_utils.py:62]: I'm sure it's not necessary LOL

File: nerfstudio/configs/method_configs.py [hunk 3]
```diff
@@ -358,7 +355,7 @@
     max_num_iterations=30000,
     mixed_precision=True,
     pipeline=VanillaPipelineConfig(
-        datamanager=VariableResDataManagerConfig(  # NOTE: one of the only differences with nerfacto
+        datamanager=VanillaDataManagerConfig(
```
[the remark this answers is not in the document]
Comment (machenmusik) [reply nerfstudio/configs/method_configs.py:358]: ... actually, maybe add back VariableRes but rename to LargeDatasetDataManager?

[the remark this answers is not in the document]
Comment (machenmusik) [reply nerfstudio/configs/method_configs.py:358]: ... nvm, [7d17dd8](https://github.com/nerfstudio-project/nerfstudio/pull/1651/commits/7d17dd8c70dd448f6b17d5a42f20d7c663adf1f9)

**Review by tancik — APPROVED** [review]
Lgtm

**Pull request merged**